plugins/adrv9002.c

```diff
@@ -307,6 +307,85 @@ static void save_digital_gain_ctl(GtkWidget *widget, struct adrv9002_rx *rx)
 	g_free(digital_gain);
 }
 
+static void orx_control_track_cals(const struct adrv9002_common *chan, bool en)
+{
+	int i = 0;
+
+	if (!chan->enabled)
+		return;
+
+	for (i = 0; i < chan->num_widgets; i++) {
+		if (!strstr(chan->w[i].attr_name, "_tracking_en"))
+			continue;
+		gtk_widget_set_sensitive(chan->w[i].widget, en);
+	}
+}
+
+static void orx_control_tx_widgets_visibility(const struct adrv9002_common *tx, bool en)
+{
+	int i = 0;
+
+	/* Disable all tx controls that might affect tx ensm state if @en=true */
+	gtk_widget_set_sensitive(tx->carrier.widget, en);
+	gtk_widget_set_sensitive(tx->ensm.w.widget, en);
+	gtk_widget_set_sensitive(tx->port_en.w.widget, en);
+	gtk_widget_set_sensitive(tx->gain_ctrl.w.widget, en);
+
+	for (i = 0; i < tx->num_widgets; i++) {
+		if (strcmp(tx->w[i].attr_name, "en"))
+			continue;
+		gtk_widget_set_sensitive(tx->w[i].widget, en);
+	}
+}
+
+static void orx_control_track_cal_visibility(const struct adrv9002_orx *orx, bool en)
+{
+	int i;
+	int other = ~orx->idx & 0x1;
+	const struct adrv9002_orx *orx_other = &orx->priv->orx_widgets[other];
+	gboolean wired;
+
+	/*
+	 * The thing with tracking cals is that, due to the way the device driver API/Firmware
+	 * is designed, we need to move all 4 ports to calibrated state to change a cal
+	 * (even if we are only changing cals in one specific port). Hence, we need to make
+	 * sure that if one of the ORxs is enabled, we block all the tracking calibrations
+	 * controls in all enabled ports. The driver would not allow it anyways, so this way,
+	 * we make it explicit to the user that this is not permitted.
+	 */
+	if (!en)
+		wired = false;
+	else
+		wired = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(orx_other->orx_en.widget));
+
+	for (i = 0; i < orx->num_widgets; i++) {
+		if (strcmp(orx->w[i].attr_name, "orx_quadrature_w_poly_tracking_en"))
+			continue;
+		gtk_widget_set_sensitive(orx->w[i].widget, wired);
+		gtk_widget_set_sensitive(orx_other->w[i].widget, wired);
+		break;
+	}
+
+	/* control track calls on both RXs */
+	orx_control_track_cals(&orx->priv->rx_widgets[other].rx, wired);
+	orx_control_track_cals(&orx->priv->rx_widgets[orx->idx].rx, wired);
+	/* control track calls on both TXs */
+	orx_control_track_cals(&orx->priv->tx_widgets[other], wired);
+	orx_control_track_cals(&orx->priv->tx_widgets[orx->idx], wired);
+}
+
+static void orx_control_rx_widgets_visibility(const struct adrv9002_common *rx, bool en)
+{
+	char rx_str[32];
+	GtkWidget *rx_frame;
+
+	/* Just disable all the RX frame as it does not make sense to control if @en=true*/
+	sprintf(rx_str, "frame_rx%d", rx->idx + 1);
+	rx_frame = GTK_WIDGET(gtk_builder_get_object(rx->priv->builder, rx_str));
```

Comment:
It would be more bulletproof to check if rx_frame is not NULL before going forward.

Reply:
I can do that :+1: